tests/functional/pod_and_daemons/test_mgr_enable_rook_backend_module.py

```diff
@@ -0,0 +1,115 @@
+            toolbox.exec_ceph_cmd('ceph orch set backend ""')
+            toolbox.exec_ceph_cmd("ceph mgr module disable rook")
+
+            log.info("Validating orch status after disabling rook module")
+            try:
+                toolbox.exec_ceph_cmd(ceph_cmd="ceph orch status", format=None)
+            except CommandFailed as ecf:
+                if "No orchestrator configured" in str(ecf):
+                    log.info("Backend Mgr rook module has been successfully disabled")
+            else:
+                raise Exception("Mgr rook module is not disabled during teardown")
+            finally:
+                log.info("Archive crash if they are raised during the test")
+                crash_check = TimeoutSampler(
+                    timeout=120,
+                    sleep=10,
+                    func=run_cmd_verify_cli_output,
+                    cmd="ceph health detail",
+                    expected_output_lst={
+                        "HEALTH_WARN",
+                        "daemons have recently crashed",
+                    },
+                    cephtool_cmd=True,
+                )
+                if crash_check.wait_for_func_status(True):
+                    toolbox.exec_ceph_cmd(ceph_cmd="ceph crash archive-all")
+                else:
+                    log.info("There are no daemon crash warnings")
+
+        request.addfinalizer(finalizer)
+
+    def test_mgr_enable_rook_backend_module(self):
+        """
+        Test to verify there are no crashes post enabling rook module on mgr and
+        setting rook as an orchestrator backend.
+        BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2274165
+
+        """
+        toolbox = pod.get_ceph_tools_pod()
+
+        log.info("Enabling rook module on mgr")
+        toolbox.exec_ceph_cmd("ceph mgr module enable rook")
+        log.info("Setting orchestrator backend to rook")
+        toolbox.exec_ceph_cmd("ceph orch set backend rook")
+        log.info("Validating orch status to verify backend: rook is Available")
+        sample = TimeoutSampler(
+            timeout=120,
+            sleep=3,
+            func=run_cmd_verify_cli_output,
+            cmd="ceph orch status",
+            expected_output_lst={"rook", "Yes"},
+            cephtool_cmd=True,
+        )
+        if not sample.wait_for_func_status(True):
+            raise Exception("Rook backend module is not enabled")
+
+        mgr_pods = pod.get_mgr_pods()
+        for mgr_pod in mgr_pods:
+            log.info(f"Restarting mgr pod:{mgr_pod.name} post enabling rook module")
+            mgr_pod.delete(wait=True)
+
+        log.info(
+            "Verify that there are no crash events for any of the ceph-mgr daemons by waiting upto 600s"
+        )
+        sample = TimeoutSampler(
+            timeout=600,
+            sleep=30,
+            func=run_cmd_verify_cli_output,
+            cmd="ceph crash ls",
+            expected_output_lst={"mgr"},
```

Comment:
I think archived crashes will still be listed in the "ceph crash ls" output. The test can fail falsely if there are any existing archived mgr crashes. Instead of checking for a specific mgr, should we check for 
"expected_output_lst={"HEALTH_WARN", "daemons have recently crashed"}," and fail the test if we see this message?

Reply:
ack, make sense. I changed to use `crash ls-new` instead of `crash ls` which suits here.
As per ceph docs: `Archive a crash report so that it is no longer considered for the RECENT_CRASH health check and does not appear in the crash ls-new output (it will still appear in the crash ls output).`
https://docs.ceph.com/en/quincy/mgr/crash/#commands

Reply:
made changes as discussed.